### ISSUE TITLE
Using HW Button to make Alexa listen

### ISSUE THREAD

**Issue by lauderdice** [alexa/alexa-avs-sample-app#904]
Hi, i would like to connect a button to my Raspberry Pi and be able to trigger the "Tap to speak to Alexa" button in Javaclient with my HW button.

Does anyone know which code I need to change in the library to achieve this?

Thanks

**Comment by sanjayrd** [comment]
Hi @lauderdice,

The core SDK itself doesn't come with any component to control hardware buttons. The general flow is that audio data should be written to a SharedDataStream, which can then be passed along to the AudioInputProcessor whenever an interaction is started.

However, the Sample App does use PortAudio to record user audio data and write it to an internal buffer. This buffer is then passed along to the AudioInputProcessor whenever an interaction begins The code that does this is located mainly in `SampleApp/src/InteractionManager.cpp`, as well as in `SampleApp/src/PortAudioMicrophoneWrapper.cpp`, and `SampleApp/src/SampleApplication.cpp`. 

Hope that clears things up,
Sanjay

**Comment by lauderdice** [comment]
I don’t think you undestood what I want exactly. My theory simply is, that if you press "Tap to speak to Alexa button", some function is called and you can start speaking to Alexa. What I want, is to call this function using my HW button.

I am not really skilled in Java but I think the function is "listenViewController.listenButtonPressed()" located in /samples/javaclient......../avs/ui/controllers. Is that correct ? :)

**Comment by kencecka** [comment]
Hi @laudrdice,

Sorry for the confusion.  We now have to sample applications - the Java sample application you are using, and a C++ SDK.  Sanjay was referring to the C++ SDK.

I don't have a lot of specific direction to give, as the exact implementation will depend on how you integrate with your hardware button, but you are on the right track with listenViewController.listenButtonPressed().  Normally, the GUI button would call that function (GraphicalListenView.java), but in your case, you want to call it from your hardware button code.

Ken
